src/install-pnpm/run.ts

```diff
@@ -37,12 +37,78 @@ export async function runSelfInstaller(inputs: Inputs): Promise<number> {
   return exitCode
 }
 
+// Nearly identical to the function `actions/setup-node` uses.
+// See https://github.com/actions/setup-node/blob/39370e3970a6d050c480ffad4ff0ed4d3fdee5af/src/util.ts#L8
+function getPnpmVersionFromFile(versionFilePath: string) {
+  if (!existsSync(versionFilePath)) {
+    throw new Error(
+      `The specified pnpm version file at: ${versionFilePath} does not exist`
+    )
+  }
+
+  const contents = readFileSync(versionFilePath, 'utf8')
+
+  // Try parsing the file as a `package.json` file.
+  try {
+    const manifest = JSON.parse(contents)
+
+    // Presume package.json file.
+    if (typeof manifest === 'object' && !!manifest) {
+      // Support Volta.
+      // See https://docs.volta.sh/guide/understanding#managing-your-project
+      if (manifest.volta?.pnpm) {
+        return manifest.volta.pnpm as string
+      }
+
+      if (manifest.engines?.pnpm) {
+        return manifest.engines.pnpm as string
+      }
```

Comment:
```suggestion
      if (typeof manifest.volta?.pnpm === 'string') {
        return manifest.volta.pnpm
      }

      if (typeof manifest.engines?.pnpm === 'string') {
        return manifest.engines.pnpm
      }
```

Reply:
Also, what if both `manifest.engines.volta.pnpm` and `manifest.engines.pnpm` are specified but different from each other? Shouldn't it throw an error?